### ISSUE TITLE
Angular 9 Angular Dependency Injection issue

### ISSUE THREAD

**Issue by Milan03** [Stabzs/Angular2-Toaster#200]
Not sure if anyone else is experiencing this but same config that worked for 8.2 isn't working on Angular 9.0.2, I get the following error when navigating to the page that invokes the ToasterService:

```
ERROR Error: Uncaught (in promise): Error: This constructor is not compatible with Angular Dependency Injection because its dependency at index 0 of the parameter list is invalid.
This can happen if the dependency type is a primitive like a string or if an ancestor of this class is missing an Angular decorator.

Please check that 1) the type for the parameter at index 0 is correct and 2) the correct Angular decorators are defined for this class and its ancestors.
Error: This constructor is not compatible with Angular Dependency Injection because its dependency at index 0 of the parameter list is invalid.
This can happen if the dependency type is a primitive like a string or if an ancestor of this class is missing an Angular decorator.

Please check that 1) the type for the parameter at index 0 is correct and 2) the correct Angular decorators are defined for this class and its ancestors.
    at ɵɵinvalidFactoryDep (core.js:939)
    at NodeInjectorFactory.ToasterContainerComponent_Factory [as factory] (ɵfac.js? [sm]:1)
    at getNodeInjectable (core.js:5643)
    at instantiateAllDirectives (core.js:12734)
    at createDirectivesInstances (core.js:11970)
    at ɵɵelementStart (core.js:21033)
    at Module.ɵɵelement (core.js:21089)
    at InboxComponent_Template (inbox.component.html:5)
    at executeTemplate (core.js:11930)
    at renderView (core.js:11716)
    at resolvePromise (zone-evergreen.js:793)
    at resolvePromise (zone-evergreen.js:752)
    at zone-evergreen.js:854
    at ZoneDelegate.invokeTask (zone-evergreen.js:400)
    at Object.onInvokeTask (core.js:40888)
    at ZoneDelegate.invokeTask (zone-evergreen.js:399)
    at Zone.runTask (zone-evergreen.js:168)
    at drainMicroTaskQueue (zone-evergreen.js:570)
```

I have a SharedModule that imports the `ToasterModule.forRoot()` and exports it. In my InboxModule, I import SharedModule and invoke the ToasterService there. This was working in all previous versions of Angular and now seems to be broken.

**Comment by ajayambre** [comment]
@Milan03 I am able to use this with Angular 9.
The error log says: `This can happen if the dependency type is a primitive like a string or if an ancestor of this class is missing an Angular decorator.`

It also suggests a way to correct this: `Please check that 1) the type for the parameter at index 0 is correct and 2) the correct Angular decorators are defined for this class and its ancestors.`

With Angular 9, they made it mandatory to mark the parent classes(without their own template) with `@Directive` if they have any `@Input/@Output` or used any other Angular decorators.

**Comment by Stabzs** [comment]
The library is now re-published under version 10.0.0 pinned against Angular 10.  I won't be going back to republish a 9.0.0 version of the library so I'll be closing this issue.
